doc/development/deprecations.rst

```diff
@@ -9,7 +9,7 @@ Pending deprecations
 * ``qml.ExpvalCost`` has been deprecated, and usage will now raise a warning.
   
   - Deprecated in v0.24
-  - Will be removed in v0.32
+  - Will be removed in v0.35
```

Comment:
If this is the case we should probably update `ShotAdaptiveOptimizer` for 0.34. A big problem with the optimizer was that when using `ExpvalCost`, it would split each Hamiltonian term into its own QNode and allocate shots. Now that we have shots on the tape, we can update this behaviour.

Reply:
Yep agreed, `ShotAdaptiveOptimizer` is in the 0.34 plan.